pkg/cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -121,8 +121,12 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 	}
 
 	defer client.Close()
-
-	return deleteVolumeAndDomain(machine, client)
+	exists, err := libvirtutils.DomainExists(machine.Name, client)
```

Comment:
This may get evaluated incorrectly in case there are two namespaces with identical machine names. The existence test needs to take the namespace into account as well.

Reply:
@enxebre looks like we completely ignore the namespace https://github.com/openshift/cluster-api-provider-libvirt/blob/12a25c9b1c1cf88311d518737c7363d260f8b5bc/pkg/cloud/libvirt/actuators/machine/actuator.go#L184

Reply:
you mean k8s namespaces? 
Though i dont think it is a valid use case to have machine objects in different namespaces, for example installer creates machines only in openshift-cluster-api namespace. Still if we want to secure this scenario, then we should create domain name using a combination of name and namespace throughout the actuator. I think that should be handled in a seperate PR.

Reply:
That's a valid argument. In that case we should search for a domain based on UUID instead.

Reply:
we cant search using UUID because UUID is not used while creating the domain. IMHO, this "uniqueness" should be handled in a later PR.